src/lib.rs

```diff
@@ -137,6 +138,39 @@ impl AtomicF64 {
     }
 }
 
+/// Utility functions for arguments sanity check
+pub(crate) fn assert_is_valid_sample_rate(sample_rate: SampleRate) {
```

Comment:
I introduced these function, I thought it could be nice to have more of them to harmonize checks and errors messages (thinking about AudioParams timing arguments that should be checked at some point). Let me know if you agree with that, can be perfectly be moved in `buffer.rs`.

I tried to add them in `ConcreteBaseAudioContext::new` but that was crashing many tests, maybe for another PR if you think we should continue in this direction.

Reply:
Looks great, let's do it this way indeed.

Few remarks:
- naming: `assert_is_valid_sample_rate` --> `assert_valid_sample_rate` (same for the others)
- could you document the panics in the docblock, so all callers can copy that in their docblock
- I think `8000 to 96000` is needlessly strict. We only need to guard for `>0` to prevent crashes, so let's do only that?

Reply:
ok done, I have added a comment about "real-life" sample rates in the doc block